### ISSUE TITLE
Make the authentication flag (-a) optional

### ISSUE THREAD

**Issue by MatthewKosloski** [MatthewKosloski/sfdc-injector#1]
If it exists, authentication arguments are read from `App.config`, and these serve as the default and can be optionally overridden by args supplied to the CLI.  In other words, if there is a complete `App.config`, no authentication arguments need to be provided to the CLI under the `-a` flag.  

However the `-a` flag is still required with blank CLI args (`-a ::::`).

Thus, to indicate an intention to use the authentication arguments from `App.config`, you can do:

```bash
dotnet SFDCInjector.dll inject -a :::: -e EventClassName:EventFieldPropertyValue
```

To improve user experience, I propose that the `-a` flag should be optional and only included if there is an intention to override one or more authentication arguments from `App.config`.

In other words, I propose that aforementioned usage of the injector be simplified to:

```bash
dotnet SFDCInjector.dll inject -e EventClassName:EventFieldPropertyValue
```
